smac/intensification/successive_halving.py

```diff
@@ -0,0 +1,475 @@
+import time
+import logging
+import typing
+from collections import OrderedDict
+from itertools import islice
+import warnings
+
+import numpy as np
+
+from smac.intensification.intensification import Intensifier
+from smac.optimizer.ei_optimization import ChallengerList
+from smac.optimizer.objective import sum_cost
+from smac.stats.stats import Stats
+from smac.utils.constants import MAXINT
+from smac.configspace import Configuration
+from smac.runhistory.runhistory import RunHistory, InstSeedKey
+from smac.tae.execute_ta_run import BudgetExhaustedException, CappedRunException, ExecuteTARun
+from smac.utils.io.traj_logging import TrajLogger
+
+__author__ = "Ashwin Raaghav Narayanan"
+__copyright__ = "Copyright 2018, ML4AAD"
+__license__ = "3-clause BSD"
+
+
+class SuccessiveHalving(Intensifier):
```

Comment:
`Intensifier` might not be a good base class to inherit from. Please create an Abstract Racing class and let `Intensifier` and `SuccessiveHalving` inherit from that

Reply:
This will be fixed in a future PR.